.github/actions/deploy/action.yml

```diff
@@ -66,9 +66,9 @@ runs:
           ${{ env.key_vault_infra_secret_name }}
 
     - name: Pin Terraform version
-      uses: hashicorp/setup-terraform@v1.2.1
+      uses: hashicorp/setup-terraform@v2.0.0
```

Comment:
We're thinking that we should only pin the major version here i.e. v2
This keeps it the same as per other steps in the workflow